### PR TITLE
Implement query() method

### PR DESCRIPTION
**Pull request by PaulKovalov**
Here I implement the basic part of week 5 task, the query method. All tests are passing

tldr:
You have a team, and each team member have some events in their cal, write a method, that will take all team's events, a list of people who must participate in the new event, and required duration of the new event, and will return all possible time slots so that every team member can attend.

Full story: https://github.com/googleinterns/step/blob/master/walkthroughs/week-5-tdd/project/calendar-walkthrough.md

As there are no clarifications about the input data, I assumed that events are not sorted.

### CODE REVIEW

**Comment by NeilLi1992** [comment]
Should we add more tests to `FindMeetingQueryTest.java` as in the instruction?

**Comment by PaulKovalov** [comment]
> Should we add more tests to `FindMeetingQueryTest.java` as in the instruction?

yes, for now I've just implemented `query`

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,101 @@
...
+    // another corner case when there is may be a gap between the end of the day and the last interval
+    Interval last = takenIntervals.get(takenIntervals.size() - 1);
+    if (last.end != 1440) {
+      // add only if enough duration
+      if (1440 - last.end >= request.getDuration()) {
+        answer.add(TimeRange.fromStartEnd(last.end, 1440, false));
+      }
+    }
+    return answer;
+  }
```
Comment (NeilLi1992) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:108]: We can eliminate the corner cases by inserting two dummy intervals at (0,0) & (1440, 1440) right?

Comment (PaulKovalov) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:108]: Great idea!

**Pull request merged**